### PR TITLE
merge develop to rc2

### PR DESCRIPTION
**Pull request by dhanendra06**

### CODE REVIEW

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java [hunk 1]
```diff
@@ -83,30 +82,25 @@ public Object getApi(ApiName apiName, List<String> pathsegments, String queryPar
 			try {
 
 				uriComponents = builder.build(false).encode();
-				printLogger.debug(uriComponents.toUri().toString(), "URI", "", "");
+				printLogger.debug(uriComponents.toUri().toString()+ "URI");
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java:85]: Do not use + to concatenate

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java [hunk 2]
```diff
@@ -83,30 +82,25 @@ public Object getApi(ApiName apiName, List<String> pathsegments, String queryPar
...
 				obj = restApiClient.getApi(uriComponents.toUri(), responseType);
 
 			} catch (Exception e) {
 				e.printStackTrace();
-				printLogger.error(LoggerFileConstant.SESSIONID.toString(),
-						LoggerFileConstant.REGISTRATIONID.toString(), "",
-						e.getMessage() + ExceptionUtils.getStackTrace(e));
-
+				printLogger.error(e.getMessage() + ExceptionUtils.getStackTrace(e));
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java:90]: Do not use concatenation. To print stackstrace use as mentioned here. 
`printLogger.error(e.getMessage(), e);`

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java [hunk 3]
```diff
@@ -83,30 +82,25 @@ public Object getApi(ApiName apiName, List<String> pathsegments, String queryPar
 			try {
 
 				uriComponents = builder.build(false).encode();
-				printLogger.debug(uriComponents.toUri().toString(), "URI", "", "");
+				printLogger.debug(uriComponents.toUri().toString()+ "URI");
 				obj = restApiClient.getApi(uriComponents.toUri(), responseType);
 
 			} catch (Exception e) {
 				e.printStackTrace();
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java:89]: Do not use printStackTrace anywhere in your code. Remove them in all places.

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java [hunk 4]
```diff
@@ -134,29 +128,25 @@ public Object getApi(ApiName apiName, List<String> pathsegments, List<String> qu
 			try {
 
 				uriComponents = builder.build(false).encode();
-				printLogger.debug(uriComponents.toUri().toString(), "URI", "", "");
+				printLogger.debug(uriComponents.toUri().toString()+"URI");
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java:131]: Do not use + mark to concatenate. Use ,

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java [hunk 5]
```diff
@@ -134,29 +128,25 @@ public Object getApi(ApiName apiName, List<String> pathsegments, List<String> qu
...
 				obj = restApiClient.getApi(uriComponents.toUri(), responseType);
 
 			} catch (Exception e) {
 				e.printStackTrace();
-				printLogger.error(LoggerFileConstant.SESSIONID.toString(),
-						LoggerFileConstant.REGISTRATIONID.toString(), "",
-						e.getMessage() + ExceptionUtils.getStackTrace(e));
+				printLogger.error(e.getMessage() + ExceptionUtils.getStackTrace(e));
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java:136]: Same as earlier stack trace comment.

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java [hunk 6]
```diff
@@ -249,26 +235,22 @@ public Object postApi(ApiName apiName, List<String> pathsegments, String queryPa
 				obj = restApiClient.postApi(builder.toUriString(), null, requestedData, responseType);
 
 			} catch (Exception e) {
-				printLogger.error(LoggerFileConstant.SESSIONID.toString(),
-						LoggerFileConstant.REGISTRATIONID.toString(), "",
-						e.getMessage() + ExceptionUtils.getStackTrace(e));
+				printLogger.error(e.getMessage() + ExceptionUtils.getStackTrace(e));
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java:238]: Same as earlier stack trace comment

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java [hunk 7]
```diff
@@ -295,128 +277,18 @@ public Object postApi(ApiName apiName, MediaType mediaType, List<String> pathseg
 				obj = restApiClient.postApi(builder.toUriString(), mediaType, requestedData, responseType);
 
 			} catch (Exception e) {
-				printLogger.error(LoggerFileConstant.SESSIONID.toString(),
-						LoggerFileConstant.REGISTRATIONID.toString(), "",
-						e.getMessage() + ExceptionUtils.getStackTrace(e));
+				printLogger.error(e.getMessage() + ExceptionUtils.getStackTrace(e));
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java:280]: Same as other stack trace comments

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java [hunk 8]
```diff
@@ -224,33 +224,38 @@
...
+			String ecryptionPin = eventModel.getEvent().getData().get("protectionKey").toString();
+			decodedCrdential = cryptoCoreUtil.decrypt(credential);
+			Map proofMap = new HashMap<String, String>();
+			proofMap = (Map) eventModel.getEvent().getData().get("proof");
+			String sign = proofMap.get("signature").toString();
+			pdfbytes = getDocuments(decodedCrdential,
+					eventModel.getEvent().getData().get("credentialType").toString(), ecryptionPin,
+					eventModel.getEvent().getTransactionId(), sign, "UIN", false).get("uinPdf");
+		}catch (Exception e){
+			printLogger.error(e.getMessage() + ExceptionUtils.getStackTrace(e));
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java:247]: same as other stackstrace comment.

File: src/main/java/io/mosip/print/service/impl/UinCardGeneratorImpl.java [hunk 9]
```diff
@@ -125,22 +124,18 @@
...
 			throw new PDFGeneratorException(PDFGeneratorExceptionCodeConstant.PDF_EXCEPTION.getErrorCode(),
 					e.getMessage() + ExceptionUtils.getStackTrace(e));
 		} 
 			  catch (ApisResourceAccessException e) {
 					e.printStackTrace();
-					printLogger.error(LoggerFileConstant.SESSIONID.toString(),
-			 LoggerFileConstant.REGISTRATIONID.toString(), "",
-						PlatformErrorMessages.PRT_PRT_PDF_SIGNATURE_EXCEPTION.name() + e.getMessage()
+					printLogger.error(PlatformErrorMessages.PRT_PRT_PDF_SIGNATURE_EXCEPTION.name() + e.getMessage()
 			 + ExceptionUtils.getStackTrace(e)); throw new PDFSignatureException(
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/service/impl/UinCardGeneratorImpl.java:135]: this is not the correct way to throw.  use as below. Also, change in all places where you are throwing an exception
`throw new PDFSignatureException(e);`

**Review by gsasikumar — CHANGES_REQUESTED** [review]

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java (hunk 1 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java]: taken care

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java (hunk 2 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java]: taken care

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java (hunk 3 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java]: taken care

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java (hunk 4 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java]: taken care

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java (hunk 5 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java]: taken care

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java (hunk 6 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java]: taken care

File: src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java (hunk 7 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/service/impl/PrintRestClientServiceImpl.java]: taken care

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java (hunk 8 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java]: taken care

File: src/main/java/io/mosip/print/service/impl/UinCardGeneratorImpl.java (hunk 9 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/service/impl/UinCardGeneratorImpl.java]: taken care

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java [hunk 10]
```diff
@@ -222,35 +222,42 @@
...
-		String ecryptionPin = eventModel.getEvent().getData().get("protectionKey").toString();
-		decodedCrdential = cryptoCoreUtil.decrypt(credential);
-		Map proofMap = new HashMap<String, String>();
-		proofMap = (Map) eventModel.getEvent().getData().get("proof");
-		String sign = proofMap.get("signature").toString();
-		byte[] pdfbytes = getDocuments(decodedCrdential,
-				eventModel.getEvent().getData().get("credentialType").toString(), ecryptionPin,
-				eventModel.getEvent().getTransactionId(), getSignature(sign, credential), "UIN", false).get("uinPdf");
-		return pdfbytes;
+		return isPrinted=true;
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java:251]: ?? There is no return inside the catch block. So all the time you will return true.

File: src/main/java/io/mosip/print/util/CryptoCoreUtil.java [hunk 11]
```diff
@@ -57,25 +64,34 @@
...
-		InputStream in = getClass().getClassLoader().getResourceAsStream("partner.p12");
-		mosipKeyStore.load(in, "password@123".toCharArray());
-		ProtectionParameter password = new PasswordProtection("password@123".toCharArray());
-		PrivateKeyEntry privateKeyEntry = (PrivateKeyEntry) mosipKeyStore.getEntry("partner", password);
+	public PrivateKeyEntry loadP12(){
+		PrivateKeyEntry privateKeyEntry = null;
+		try {
+			KeyStore mosipKeyStore = KeyStore.getInstance("PKCS12");
+			InputStream in = getClass().getClassLoader().getResourceAsStream("partner.p12");
+			mosipKeyStore.load(in, "password@123".toCharArray());
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/util/CryptoCoreUtil.java:85]: password should not be hard coded.

File: src/main/java/io/mosip/print/util/CryptoCoreUtil.java [hunk 12]
```diff
@@ -57,25 +64,34 @@
...
-		KeyStore mosipKeyStore = KeyStore.getInstance("PKCS12");
-		InputStream in = getClass().getClassLoader().getResourceAsStream("partner.p12");
-		mosipKeyStore.load(in, "password@123".toCharArray());
-		ProtectionParameter password = new PasswordProtection("password@123".toCharArray());
-		PrivateKeyEntry privateKeyEntry = (PrivateKeyEntry) mosipKeyStore.getEntry("partner", password);
+	public PrivateKeyEntry loadP12(){
+		PrivateKeyEntry privateKeyEntry = null;
+		try {
+			KeyStore mosipKeyStore = KeyStore.getInstance("PKCS12");
+			InputStream in = getClass().getClassLoader().getResourceAsStream("partner.p12");
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/util/CryptoCoreUtil.java:84]: filename should not be hard coded

File: src/main/java/io/mosip/print/controller/Print.java [hunk 13]
```diff
@@ -1,54 +1,53 @@
...
+	 * @return
+	 * @throws Exception
+	 */
+	@PostMapping(path = "/callback/notifyPrint", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+	@PreAuthenticateContentAndVerifyIntent(secret = "${mosip.event.secret}", callback = "/v1/print/print/callback/notifyPrint", topic = "${mosip.event.topic}")
+	public ResponseEntity<String> handleSubscribeEvent(@RequestBody EventModel eventModel) throws Exception {
+		printLogger.info("event recieved from websub"+", id: {}",eventModel.getEvent().getId());
+		boolean isPrinted = printService.generateCard(eventModel);
+		printLogger.info("printing status : {} for event id: {}",isPrinted,eventModel.getEvent().getId());
+		return new ResponseEntity<>("successfully printed", HttpStatus.OK);
```
Comment (gsasikumar) [inline src/main/java/io/mosip/print/controller/Print.java:50]: Request accepted.

**Review by gsasikumar — CHANGES_REQUESTED** [review]

File: src/main/java/io/mosip/print/controller/Print.java (hunk 13 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/controller/Print.java]: taken care

File: src/main/java/io/mosip/print/util/CryptoCoreUtil.java (hunk 12 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/util/CryptoCoreUtil.java]: taken care

File: src/main/java/io/mosip/print/util/CryptoCoreUtil.java (hunk 11 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/util/CryptoCoreUtil.java:95]: taken care

File: src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java (hunk 10 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/print/service/impl/PrintServiceImpl.java:252]: taken care

**Review by gsasikumar — APPROVED** [review]

**Pull request merged**